### ISSUE TITLE
Fail error: "list" argument must be an Array of Buffers

### ISSUE THREAD

**Issue by dustinbolton** [stream-utils/duplex-child-process#8]
It looks like the response is unexpected so it's crashing. Also not sure what's going on with the negative 2 exit code from the command I'm trying to run.

```
buffer.js:296
    throw new TypeError('"list" argument must be an Array of Buffers');
    ^

TypeError: "list" argument must be an Array of Buffers
    at Function.Buffer.concat (buffer.js:296:11)
    at ChildProcess.onExit (/var/app/current/node_modules/duplex-child-process/index.js:123:50)
    at ChildProcess.g (events.js:291:16)
    at emitTwo (events.js:106:13)
    at ChildProcess.emit (events.js:191:7)
    at maybeClose (internal/child_process.js:877:16)
    at Socket.<anonymous> (internal/child_process.js:334:11)
    at emitOne (events.js:96:13)
    at Socket.emit (events.js:188:7)
    at Pipe._handle.close [as _onclose] (net.js:498:12)
```

I output the contents of stderr, code, and signal variables to console to help troubleshoot:
```
stderr:
null
code:
-2
signal:
null
```

**Comment by jeromew** [comment]
Sorry for the delay. Can you tell me what command & options you are trying to spawn ?

It seems that the module is a bit aggressive for some linux commands that have non-blocking errors.

I am currently trying to debug a similar issue while using wget.

the error you got seem to come from the fact that the onExit function is called several times unexpectedly.

**Comment by jeromew** [comment]
@overlookmotel the module works in my use case with node 7.10 but there may be some tests that are failing. I'll try to have a look in the following weeks.

this module brings a nice syntax to handle external process as streams but it needs more users and tests to discover how the assumptions it makes on some event sequences work in all cases. External processes can sometimes behave in unexpected ways wrt their pipes.

**Comment by jeromew** [comment]
I think this is fixed with 1.0.0. onExit could be called several times and the guard condition was wrong.
